File: naked-generator/Secondary/py/fixed_ids.py
from enum import Enum
from struct import pack, unpack
from collections import namedtuple


IMU_ANGULAR_RATE = namedtuple('IMU_ANGULAR_RATE', 'ang_rate_x ang_rate_y ang_rate_z')
IMU_ANGULAR_RATE_schema = '<HHH'

IMU_ACCELERATION = namedtuple('IMU_ACCELERATION', 'accel_x accel_y accel_z')
IMU_ACCELERATION_schema = '<HHH'

IRTS_FL_0 = namedtuple('IRTS_FL_0', 'channel1 channel2 channel3 channel4')
IRTS_FL_0_schema = '<HHHH'

IRTS_FL_1 = namedtuple('IRTS_FL_1', 'channel5 channel6 channel7 channel8')
IRTS_FL_1_schema = '<HHHH'

IRTS_FL_2 = namedtuple('IRTS_FL_2', 'channel9 channel10 channel11 channel12')
IRTS_FL_2_schema = '<HHHH'

IRTS_FL_3 = namedtuple('IRTS_FL_3', 'channel13 channel14 channel15 channel16')
IRTS_FL_3_schema = '<HHHH'

IRTS_FR_0 = namedtuple('IRTS_FR_0', 'channel1 channel2 channel3 channel4')
IRTS_FR_0_schema = '<HHHH'

IRTS_FR_1 = namedtuple('IRTS_FR_1', 'channel5 channel6 channel7 channel8')
IRTS_FR_1_schema = '<HHHH'

IRTS_FR_2 = namedtuple('IRTS_FR_2', 'channel9 channel10 channel11 channel12')
IRTS_FR_2_schema = '<HHHH'

IRTS_FR_3 = namedtuple('IRTS_FR_3', 'channel13 channel14 channel15 channel16')
IRTS_FR_3_schema = '<HHHH'

IRTS_RL_0 = namedtuple('IRTS_RL_0', 'channel1 channel2 channel3 channel4')
IRTS_RL_0_schema = '<HHHH'

IRTS_RL_1 = namedtuple('IRTS_RL_1', 'channel5 channel6 channel7 channel8')
IRTS_RL_1_schema = '<HHHH'

IRTS_RL_2 = namedtuple('IRTS_RL_2', 'channel9 channel10 channel11 channel12')
IRTS_RL_2_schema = '<HHHH'

IRTS_RL_3 = namedtuple('IRTS_RL_3', 'channel13 channel14 channel15 channel16')
IRTS_RL_3_schema = '<HHHH'

IRTS_RR_0 = namedtuple('IRTS_RR_0', 'channel1 channel2 channel3 channel4')
IRTS_RR_0_schema = '<HHHH'

IRTS_RR_1 = namedtuple('IRTS_RR_1', 'channel5 channel6 channel7 channel8')
IRTS_RR_1_schema = '<HHHH'

IRTS_RR_2 = namedtuple('IRTS_RR_2', 'channel9 channel10 channel11 channel12')
IRTS_RR_2_schema = '<HHHH'

IRTS_RR_3 = namedtuple('IRTS_RR_3', 'channel13 channel14 channel15 channel16')
IRTS_RR_3_schema = '<HHHH'


def serialize_IMU_ANGULAR_RATE(struct: IMU_ANGULAR_RATE) -> bytearray:
	return pack(IMU_ANGULAR_RATE, *tuple(struct))


def serialize_IMU_ACCELERATION(struct: IMU_ACCELERATION) -> bytearray:
	return pack(IMU_ACCELERATION, *tuple(struct))


def serialize_IRTS_FL_0(struct: IRTS_FL_0) -> bytearray:
	return pack(IRTS_FL_0, *tuple(struct))


def serialize_IRTS_FL_1(struct: IRTS_FL_1) -> bytearray:
	return pack(IRTS_FL_1, *tuple(struct))


def serialize_IRTS_FL_2(struct: IRTS_FL_2) -> bytearray:
	return pack(IRTS_FL_2, *tuple(struct))


def serialize_IRTS_FL_3(struct: IRTS_FL_3) -> bytearray:
	return pack(IRTS_FL_3, *tuple(struct))


def serialize_IRTS_FR_0(struct: IRTS_FR_0) -> bytearray:
	return pack(IRTS_FR_0, *tuple(struct))


def serialize_IRTS_FR_1(struct: IRTS_FR_1) -> bytearray:
	return pack(IRTS_FR_1, *tuple(struct))


def serialize_IRTS_FR_2(struct: IRTS_FR_2) -> bytearray:
	return pack(IRTS_FR_2, *tuple(struct))


def serialize_IRTS_FR_3(struct: IRTS_FR_3) -> bytearray:
	return pack(IRTS_FR_3, *tuple(struct))


def serialize_IRTS_RL_0(struct: IRTS_RL_0) -> bytearray:
	return pack(IRTS_RL_0, *tuple(struct))


def serialize_IRTS_RL_1(struct: IRTS_RL_1) -> bytearray:
	return pack(IRTS_RL_1, *tuple(struct))


def serialize_IRTS_RL_2(struct: IRTS_RL_2) -> bytearray:
	return pack(IRTS_RL_2, *tuple(struct))


def serialize_IRTS_RL_3(struct: IRTS_RL_3) -> bytearray:
	return pack(IRTS_RL_3, *tuple(struct))


def serialize_IRTS_RR_0(struct: IRTS_RR_0) -> bytearray:
	return pack(IRTS_RR_0, *tuple(struct))


def serialize_IRTS_RR_1(struct: IRTS_RR_1) -> bytearray:
	return pack(IRTS_RR_1, *tuple(struct))


def serialize_IRTS_RR_2(struct: IRTS_RR_2) -> bytearray:
	return pack(IRTS_RR_2, *tuple(struct))


def serialize_IRTS_RR_3(struct: IRTS_RR_3) -> bytearray:
	return pack(IRTS_RR_3, *tuple(struct))


def deserialize_IMU_ANGULAR_RATE(buffer: bytearray) -> IMU_ANGULAR_RATE:
	return IMU_ANGULAR_RATE._make(unpack(IMU_ANGULAR_RATE_schema, buffer))


def deserialize_IMU_ACCELERATION(buffer: bytearray) -> IMU_ACCELERATION:
	return IMU_ACCELERATION._make(unpack(IMU_ACCELERATION_schema, buffer))


def deserialize_IRTS_FL_0(buffer: bytearray) -> IRTS_FL_0:
	return IRTS_FL_0._make(unpack(IRTS_FL_0_schema, buffer))


def deserialize_IRTS_FL_1(buffer: bytearray) -> IRTS_FL_1:
	return IRTS_FL_1._make(unpack(IRTS_FL_1_schema, buffer))


def deserialize_IRTS_FL_2(buffer: bytearray) -> IRTS_FL_2:
	return IRTS_FL_2._make(unpack(IRTS_FL_2_schema, buffer))


def deserialize_IRTS_FL_3(buffer: bytearray) -> IRTS_FL_3:
	return IRTS_FL_3._make(unpack(IRTS_FL_3_schema, buffer))


def deserialize_IRTS_FR_0(buffer: bytearray) -> IRTS_FR_0:
	return IRTS_FR_0._make(unpack(IRTS_FR_0_schema, buffer))


def deserialize_IRTS_FR_1(buffer: bytearray) -> IRTS_FR_1:
	return IRTS_FR_1._make(unpack(IRTS_FR_1_schema, buffer))


def deserialize_IRTS_FR_2(buffer: bytearray) -> IRTS_FR_2:
	return IRTS_FR_2._make(unpack(IRTS_FR_2_schema, buffer))


def deserialize_IRTS_FR_3(buffer: bytearray) -> IRTS_FR_3:
	return IRTS_FR_3._make(unpack(IRTS_FR_3_schema, buffer))


def deserialize_IRTS_RL_0(buffer: bytearray) -> IRTS_RL_0:
	return IRTS_RL_0._make(unpack(IRTS_RL_0_schema, buffer))


def deserialize_IRTS_RL_1(buffer: bytearray) -> IRTS_RL_1:
	return IRTS_RL_1._make(unpack(IRTS_RL_1_schema, buffer))


def deserialize_IRTS_RL_2(buffer: bytearray) -> IRTS_RL_2:
	return IRTS_RL_2._make(unpack(IRTS_RL_2_schema, buffer))


def deserialize_IRTS_RL_3(buffer: bytearray) -> IRTS_RL_3:
	return IRTS_RL_3._make(unpack(IRTS_RL_3_schema, buffer))


def deserialize_IRTS_RR_0(buffer: bytearray) -> IRTS_RR_0:
	return IRTS_RR_0._make(unpack(IRTS_RR_0_schema, buffer))


def deserialize_IRTS_RR_1(buffer: bytearray) -> IRTS_RR_1:
	return IRTS_RR_1._make(unpack(IRTS_RR_1_schema, buffer))


def deserialize_IRTS_RR_2(buffer: bytearray) -> IRTS_RR_2:
	return IRTS_RR_2._make(unpack(IRTS_RR_2_schema, buffer))


def deserialize_IRTS_RR_3(buffer: bytearray) -> IRTS_RR_3:
	return IRTS_RR_3._make(unpack(IRTS_RR_3_schema, buffer))

File: naked-generator/Secondary/c/fixed_ids.c
#include <string.h>
#include <assert.h>
#include <stdio.h>

#include "fixed_ids.h"

void serialize_IMU_ANGULAR_RATE(IMU_ANGULAR_RATE* imu_angular_rate, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IMU_ANGULAR_RATE));
	memcpy(buffer, imu_angular_rate, sizeof(STEER_STATUS));
}
void serialize_IMU_ACCELERATION(IMU_ACCELERATION* imu_acceleration, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IMU_ACCELERATION));
	memcpy(buffer, imu_acceleration, sizeof(STEER_STATUS));
}
void serialize_IRTS_FL_0(IRTS_FL_0* irts_fl_0, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FL_0));
	memcpy(buffer, irts_fl_0, sizeof(STEER_STATUS));
}
void serialize_IRTS_FL_1(IRTS_FL_1* irts_fl_1, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FL_1));
	memcpy(buffer, irts_fl_1, sizeof(STEER_STATUS));
}
void serialize_IRTS_FL_2(IRTS_FL_2* irts_fl_2, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FL_2));
	memcpy(buffer, irts_fl_2, sizeof(STEER_STATUS));
}
void serialize_IRTS_FL_3(IRTS_FL_3* irts_fl_3, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FL_3));
	memcpy(buffer, irts_fl_3, sizeof(STEER_STATUS));
}
void serialize_IRTS_FR_0(IRTS_FR_0* irts_fr_0, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FR_0));
	memcpy(buffer, irts_fr_0, sizeof(STEER_STATUS));
}
void serialize_IRTS_FR_1(IRTS_FR_1* irts_fr_1, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FR_1));
	memcpy(buffer, irts_fr_1, sizeof(STEER_STATUS));
}
void serialize_IRTS_FR_2(IRTS_FR_2* irts_fr_2, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FR_2));
	memcpy(buffer, irts_fr_2, sizeof(STEER_STATUS));
}
void serialize_IRTS_FR_3(IRTS_FR_3* irts_fr_3, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_FR_3));
	memcpy(buffer, irts_fr_3, sizeof(STEER_STATUS));
}
void serialize_IRTS_RL_0(IRTS_RL_0* irts_rl_0, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RL_0));
	memcpy(buffer, irts_rl_0, sizeof(STEER_STATUS));
}
void serialize_IRTS_RL_1(IRTS_RL_1* irts_rl_1, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RL_1));
	memcpy(buffer, irts_rl_1, sizeof(STEER_STATUS));
}
void serialize_IRTS_RL_2(IRTS_RL_2* irts_rl_2, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RL_2));
	memcpy(buffer, irts_rl_2, sizeof(STEER_STATUS));
}
void serialize_IRTS_RL_3(IRTS_RL_3* irts_rl_3, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RL_3));
	memcpy(buffer, irts_rl_3, sizeof(STEER_STATUS));
}
void serialize_IRTS_RR_0(IRTS_RR_0* irts_rr_0, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RR_0));
	memcpy(buffer, irts_rr_0, sizeof(STEER_STATUS));
}
void serialize_IRTS_RR_1(IRTS_RR_1* irts_rr_1, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RR_1));
	memcpy(buffer, irts_rr_1, sizeof(STEER_STATUS));
}
void serialize_IRTS_RR_2(IRTS_RR_2* irts_rr_2, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RR_2));
	memcpy(buffer, irts_rr_2, sizeof(STEER_STATUS));
}
void serialize_IRTS_RR_3(IRTS_RR_3* irts_rr_3, uint8_t* buffer, size_t buf_len) {
	assert(buf_len >= sizeof(IRTS_RR_3));
	memcpy(buffer, irts_rr_3, sizeof(STEER_STATUS));
}

void deserialize_IMU_ANGULAR_RATE(uint8_t* buffer, size_t buf_len, IMU_ANGULAR_RATE* imu_angular_rate) {
	assert(buf_len >= sizeof(IMU_ANGULAR_RATE));
	memcpy(imu_angular_rate, buffer, sizeof(STEER_STATUS));
}
void deserialize_IMU_ACCELERATION(uint8_t* buffer, size_t buf_len, IMU_ACCELERATION* imu_acceleration) {
	assert(buf_len >= sizeof(IMU_ACCELERATION));
	memcpy(imu_acceleration, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FL_0(uint8_t* buffer, size_t buf_len, IRTS_FL_0* irts_fl_0) {
	assert(buf_len >= sizeof(IRTS_FL_0));
	memcpy(irts_fl_0, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FL_1(uint8_t* buffer, size_t buf_len, IRTS_FL_1* irts_fl_1) {
	assert(buf_len >= sizeof(IRTS_FL_1));
	memcpy(irts_fl_1, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FL_2(uint8_t* buffer, size_t buf_len, IRTS_FL_2* irts_fl_2) {
	assert(buf_len >= sizeof(IRTS_FL_2));
	memcpy(irts_fl_2, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FL_3(uint8_t* buffer, size_t buf_len, IRTS_FL_3* irts_fl_3) {
	assert(buf_len >= sizeof(IRTS_FL_3));
	memcpy(irts_fl_3, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FR_0(uint8_t* buffer, size_t buf_len, IRTS_FR_0* irts_fr_0) {
	assert(buf_len >= sizeof(IRTS_FR_0));
	memcpy(irts_fr_0, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FR_1(uint8_t* buffer, size_t buf_len, IRTS_FR_1* irts_fr_1) {
	assert(buf_len >= sizeof(IRTS_FR_1));
	memcpy(irts_fr_1, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FR_2(uint8_t* buffer, size_t buf_len, IRTS_FR_2* irts_fr_2) {
	assert(buf_len >= sizeof(IRTS_FR_2));
	memcpy(irts_fr_2, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_FR_3(uint8_t* buffer, size_t buf_len, IRTS_FR_3* irts_fr_3) {
	assert(buf_len >= sizeof(IRTS_FR_3));
	memcpy(irts_fr_3, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RL_0(uint8_t* buffer, size_t buf_len, IRTS_RL_0* irts_rl_0) {
	assert(buf_len >= sizeof(IRTS_RL_0));
	memcpy(irts_rl_0, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RL_1(uint8_t* buffer, size_t buf_len, IRTS_RL_1* irts_rl_1) {
	assert(buf_len >= sizeof(IRTS_RL_1));
	memcpy(irts_rl_1, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RL_2(uint8_t* buffer, size_t buf_len, IRTS_RL_2* irts_rl_2) {
	assert(buf_len >= sizeof(IRTS_RL_2));
	memcpy(irts_rl_2, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RL_3(uint8_t* buffer, size_t buf_len, IRTS_RL_3* irts_rl_3) {
	assert(buf_len >= sizeof(IRTS_RL_3));
	memcpy(irts_rl_3, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RR_0(uint8_t* buffer, size_t buf_len, IRTS_RR_0* irts_rr_0) {
	assert(buf_len >= sizeof(IRTS_RR_0));
	memcpy(irts_rr_0, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RR_1(uint8_t* buffer, size_t buf_len, IRTS_RR_1* irts_rr_1) {
	assert(buf_len >= sizeof(IRTS_RR_1));
	memcpy(irts_rr_1, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RR_2(uint8_t* buffer, size_t buf_len, IRTS_RR_2* irts_rr_2) {
	assert(buf_len >= sizeof(IRTS_RR_2));
	memcpy(irts_rr_2, buffer, sizeof(STEER_STATUS));
}
void deserialize_IRTS_RR_3(uint8_t* buffer, size_t buf_len, IRTS_RR_3* irts_rr_3) {
	assert(buf_len >= sizeof(IRTS_RR_3));
	memcpy(irts_rr_3, buffer, sizeof(STEER_STATUS));
}
